### DISCUSSION TITLE
Auto-Aligning Whitespace

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/roslyn#48572] [Ideas] [answered]
It would be really nice if C#/VS would "Smartly" align tokens using whitespace.  Here's what I would suggest:

1.  Parse the current line.
2.  Parse the first non-whitespace line above it (we'll call this the "previous" line
3.  For each token, on the current line, if it is the same type of token as the corresponding token on the previous line, insert whitespace so that the tokens line up.
4.  All tokens should be left-aligned unless it is a numeric token, then it should be right-aligned.

The goal being that code that looks like this:
![image](https://user-images.githubusercontent.com/13609558/95914208-ff8c2080-0d6a-11eb-89c4-5b700d050ba1.png)

Gets reformatted as this:
![image](https://user-images.githubusercontent.com/13609558/95914260-1468b400-0d6b-11eb-9270-c2148a528890.png)

**Comment by HaloFour** [comment, marked as answer]
This feels like it belongs in [Roslyn](https://github.com/dotnet/roslyn/)?  The compiler isn't involved in auto-formatting or format preferences.
